src/components/view/DocumentLibraryView/DocumentLibraryView.tsx

```diff
@@ -0,0 +1,140 @@
+        }
+    };
+
+    const onUploadClick = () => {
+        setLoading(true);
+        MyDataHelps.startSurvey(props.uploadDocumentSurveyName);
+    }
+
+    const onFileClick = (file: SurveyUploadedFile) => {
+        //console.log(`Opening File ${file.title} ${file.fileName} ${file.type} ${file.date} SurveyResultID: ${file.surveyResultId}`);
+
+        if (props.documentViewBaseUrl) {
+            const documentDetailViewUrl = props.documentViewBaseUrl + '?surveyResultId=' + encodeURIComponent(file.surveyResultId)
+                + '&fileResultIdentifier=' + encodeURIComponent(props.fileResultIdentifier)
+                + '&typeResultIdentifier=' + encodeURIComponent(props.typeResultIdentifier)
+                + '&nameResultIdentifier=' + encodeURIComponent(props.nameResultIdentifier)
+                + '&dateResultIdentifier=' + encodeURIComponent(props.dateResultIdentifier)
+                + '&notesResultIdentifier=' + encodeURIComponent(props.notesResultIdentifier);
+            MyDataHelps.openApplication(documentDetailViewUrl);
+        }
+    }
+
+    async function initialize() {
+        setLoading(true);
+
+        if (props.preview) {
+            if (props.preview === "Preview") {
+                setFiles(getPreviewData());
+                setLoading(false);
+            }
+            return;
+        }
+
+        const params: SurveyUploadedFileQueryParameters = {
+            uploadDocumentSurveyName: props.uploadDocumentSurveyName,
+            fileResultIdentifier: props.fileResultIdentifier,
+            typeResultIdentifier: props.typeResultIdentifier,
+            nameResultIdentifier: props.nameResultIdentifier,
+            dateResultIdentifier: props.dateResultIdentifier,
+            notesResultIdentifier: props.notesResultIdentifier
+        };
+
+        queryAllSurveyFiles({ ...params }).then((uploadedFiles: SurveyUploadedFile[]) => {
+            setFiles(uploadedFiles);
+            setLoading(false);
+        });
+    }
+
+    function buildFileList(files: SurveyUploadedFile[], selectedSegment: "date" | "name" | "type") {
+        const sortedFiles = files.sort((a, b) => {
+            if (selectedSegment === "date") {
+                return a.date > b.date ? -1 : 1;
+            } else if (selectedSegment === "name") {
+                return a.title < b.title ? -1 : 1;
+            } else {
+                return a.type < b.type ? -1 : 1;
+            }
+        });
+
+        return sortedFiles.map((file: SurveyUploadedFile, index: number) =>
+            <Action key={`suf_${index}`} title={`${file.title} (${file.type})`}
+                subtitle={`${getShortDateString(file.date)} ${file.fileName}`}
+                icon={<FontAwesomeSvgIcon icon={faFileLines} />}
+                onClick={() => onFileClick(file)}></Action>
+        );
+    }
+
+    useInitializeView(() => {
+        initialize();
+    }, [], []);
+
+    return (
+        <Layout colorScheme={props.colorScheme}>
```

Comment:
You'll also need a navigation bar here to go back?  Which should likely contain the title / button as children like, say, the lab results news feed view...